config.go

```diff
@@ -118,10 +118,12 @@ func newConfig(options ...Option) config {
 		opt.Apply(&cfg)
 	}
 
-	cfg.Tracer = cfg.TracerProvider.Tracer(
-		instrumentationName,
-		trace.WithInstrumentationVersion(Version()),
-	)
+	cfg.Tracer = func() trace.Tracer {
```

Comment:
- You mentioned that caching the ref causes initialization issues. Can you share what those are for interest sake ?
- I assume creating a Tracer is cheap ?

Reply:
Yes of course. The original code will get the tracer during database initialisation and keep that reference throughout it entire lifetime. This means that you have to initialise the tracing before connecting to the database. But within our codebase, we need to access the database to get the config in order to configure the tracer, so ➡️ 🥚🐔 problem. 

This is a bit of a strong requirement to make for the otelsql library, and as the otel lib as a simple solution for that through the tracer provider, it's best to delegate that to the API, which simplifies things a lot. 

Yes, it's really cheap as the tracing provider in doing exactly the same thing, storing the current tracer and updating it when asked. The cost is merely a function call, which in that context is totally trivial https://sourcegraph.sourcegraph.com/github.com/open-telemetry/opentelemetry-go/-/blob/sdk/trace/provider.go?L149

Reply:
Ah that makes this change obvious! Thanks for the explanation 👏🏼

Reply:
I think best practice here might be to store a reference to the `TracerProvider`, while they store a `Tracer` here which isn't ideal. I wonder if it might be best, actually, to just remove the `Tracer` field entirely?